source/grid/grid_tools.cc

```diff
@@ -491,6 +482,60 @@ namespace GridTools
         return false;
       }
     };
+
+
+    /**
+     * get_coarse_mesh_description() needs to store face data for
+     * dim>1, but we can not have this code in the function, as this
+     * requires and instantiation of CellData<dim-1> or constexpr
+     * if. Use a class instead for now.
+     */
+    template <int dim>
+    class FaceDataHelper
+    {
+    public:
+      template <class FaceIteratorType>
+      void
+      add(const FaceIteratorType &face)
+      {
+        CellData<dim - 1> face_cell_data;
+        for (unsigned int vertex_n = 0;
+             vertex_n < GeometryInfo<dim>::vertices_per_face;
+             ++vertex_n)
+          face_cell_data.vertices[vertex_n] = face->vertex_index(vertex_n);
+        face_cell_data.boundary_id = face->boundary_id();
+        face_cell_data.manifold_id = face->manifold_id();
+
+        face_data.insert(face_cell_data);
+      }
+
+      void
+      copy(SubCellData &subcell_data)
+      {
+        for (const CellData<dim - 1> &face_cell_data : face_data)
+          append_face_data(face_cell_data, subcell_data);
+      }
+
+
+      std::set<CellData<dim - 1>, CellDataComparator<dim - 1>> face_data;
```

Comment:
Make this variable `private:` and add documentation that states that you're inserting data in the `add()` function and retrieve it in the other member function.